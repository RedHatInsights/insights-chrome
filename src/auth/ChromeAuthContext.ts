import { ChromeUser } from '@redhat-cloud-services/types';
import { AxiosResponse } from 'axios';
import { createContext } from 'react';
import { OfflineTokenResponse } from './offline';

export type ChromeLogin<LoginResponse = void> = (requiredScopes?: string[]) => Promise<LoginResponse>;

export type ChromeAuthContextValue<LoginResponse = void> = {
  ssoUrl: string;
  ready: boolean;
  user: ChromeUser;
  getUser: () => Promise<ChromeUser>;
  token: string;
  logoutAllTabs: (bounce?: boolean) => void;
  loginAllTabs: () => void;
  logout: () => void;
  login: ChromeLogin<LoginResponse>;
  tokenExpires: number;
  getToken: () => Promise<string>;
  getRefreshToken: () => Promise<string>;
  postbackUrl?: string;
  getOfflineToken: () => Promise<AxiosResponse<OfflineTokenResponse>>;
  doOffline: () => Promise<void>;
  reAuthWithScopes: (...scopes: string[]) => Promise<void>;
  forceRefresh: () => Promise<unknown>;
};

const blankUser: ChromeUser = {
  entitlements: {},
  identity: {
    org_id: '',
    type: '',
  },
};

const ChromeAuthContext = createContext<ChromeAuthContextValue>({
  ssoUrl: '',
  ready: false,
  logoutAllTabs: () => undefined,
  loginAllTabs: () => undefined,
  logout: () => undefined,
  login: () => Promise.resolve(),
  getToken: () => Promise.resolve(''),
  getRefreshToken: () => Promise.resolve(''),
  getOfflineToken: () =>
    Promise.resolve({
      data: {},
    } as AxiosResponse<OfflineTokenResponse>),
  doOffline: () => Promise.resolve(),
  getUser: () => Promise.resolve(blankUser),
  token: '',
  tokenExpires: 0,
  user: blankUser,
  reAuthWithScopes: () => Promise.resolve(),
  forceRefresh: () => Promise.resolve(),
});

export default ChromeAuthContext;
