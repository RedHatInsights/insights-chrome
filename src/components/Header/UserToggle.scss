.chr-c-disabled-pointer {
  cursor: pointer;
}
