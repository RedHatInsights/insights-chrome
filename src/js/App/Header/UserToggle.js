import React, { Component } from 'react';
import {
    Dropdown,
    DropdownToggle,
    KebabToggle,
    DropdownItem,
    DropdownSeparator,
    DropdownPosition
} from '@patternfly/react-core';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

function buildItems(username, isOrgAdmin, accountNumber = -1, extraItems) {
    return [
        <DropdownItem key="Username" isDisabled>
            <dl className='ins-c-dropdown-item__stack'>
                <dt className="ins-c-dropdown-item__stack--header">Username:</dt>
                <dd className="ins-c-dropdown-item__stack--value">{username}</dd>
            </dl>
        </DropdownItem>,
        <React.Fragment key="account wrapper">
            { accountNumber > -1 &&
                <DropdownItem key="Account" isDisabled>
                    <dl className='ins-c-dropdown-item__stack'>
                        <dt className="ins-c-dropdown-item__stack--header">Account Number:</dt>
                        <dd className="ins-c-dropdown-item__stack--value">{accountNumber}</dd>
                    </dl>
                </DropdownItem>
            }
        </React.Fragment>,
        <DropdownSeparator key="separator" />,
        <React.Fragment key="user management wrapper">
            { isOrgAdmin &&
                <DropdownItem
                    key="User management"
                    href={`https://www.${window.insights.chrome.isProd ? '' : 'qa.' }redhat.com/wapps/ugc/protected/usermgt/userList.html`}
                    target="_blank" rel='noopener noreferrer'>
                        User management
                </DropdownItem>
            }
        </React.Fragment>,
        <DropdownItem
            key="My Profile"
            href="https://access.redhat.com/user"
            target="_blank"
            rel='noopener noreferrer'>
                My profile
        </DropdownItem>,
        <DropdownItem
            key="logout"
            component="button"
            onClick={() => window.insights.chrome.auth.logout(true)}>
                Logout
        </DropdownItem>,
        [...extraItems]
    ];
}

export class UserToggle extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false
        };
        this.onSelect = this.onSelect.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    onSelect() {
        this.setState({ isOpen: !this.state.isOpen });
    }

    onToggle(isOpen) {
        this.setState({
            isOpen
        });
    }

    render() {
        const { isOpen } = this.state;
        const { account, isSmall, extraItems } = this.props;
        const toggle = isSmall ?
            <KebabToggle onToggle={this.onToggle} /> :
            <DropdownToggle onToggle={this.onToggle}>{account.name}</DropdownToggle>;
        return (
            <Dropdown
                position={DropdownPosition.right}
                aria-label="Overflow actions"
                widget-type="InsightsOverflowActions"
                onSelect={this.onSelect}
                toggle={toggle}
                isPlain
                isOpen={isOpen}
                dropdownItems={buildItems(account.username, account.isOrgAdmin, account.number, extraItems)}
            />
        );
    }
}

UserToggle.propTypes = {
    account: PropTypes.shape({
        number: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        name: PropTypes.string
    }),
    isSmall: PropTypes.bool,
    extraItems: PropTypes.arrayOf(PropTypes.node)
};

UserToggle.defaultProps = {
    account: {
        number: 1,
        name: 'Foo'
    },
    isSmall: false,
    extraItems: []
};

/* eslint-disable camelcase */
// TODO update this to use account_id
export default connect(({
    chrome: {
        user: {
            identity: {
                account_number: accountNumber,
                user: {
                    username,
                    first_name,
                    last_name,
                    is_org_admin
                }
            }
        }
    } }) => ({
    account: {
        number: accountNumber,
        username: username,
        isOrgAdmin: is_org_admin,
        name: `${first_name} ${last_name}`
    }
}))(UserToggle);

/* eslint-enable camelcase */
