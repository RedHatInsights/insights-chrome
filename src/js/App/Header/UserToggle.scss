.pf-c-dropdown__toggle.ins-c-toolbar__menu-user {
    display: flex;
}
