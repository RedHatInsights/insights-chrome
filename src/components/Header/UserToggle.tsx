import './UserToggle.scss';

import { Dropdown, DropdownItem, DropdownList } from '@patternfly/react-core/dist/dynamic/components/Dropdown';
import { ITLess, getEnv, isProd as isProdEnv } from '../../utils/common';
import React, { useContext, useRef, useState } from 'react';

import ChromeLink from '../ChromeLink/ChromeLink';
import { Divider } from '@patternfly/react-core/dist/dynamic/components/Divider';
import { EllipsisVIcon } from '@patternfly/react-icons/dist/dynamic/icons/ellipsis-v-icon';
import { MenuToggle } from '@patternfly/react-core/dist/dynamic/components/MenuToggle';
import QuestionCircleIcon from '@patternfly/react-icons/dist/dynamic/icons/question-circle-icon';
import { Tooltip } from '@patternfly/react-core/dist/dynamic/components/Tooltip';
import UserIcon from './UserIcon';
import classNames from 'classnames';
import messages from '../../locales/Messages';
import { useIntl } from 'react-intl';
import ChromeAuthContext from '../../auth/ChromeAuthContext';
import { useFlag } from '@unleash/proxy-client-react';

const DropdownItems = ({
  username = '',
  isOrgAdmin,
  accountNumber,
  orgId,
  isInternal,
  extraItems = [],
}: {
  username?: string;
  isOrgAdmin?: boolean;
  accountNumber?: string;
  orgId?: string;
  isInternal?: boolean;
  extraItems?: React.ReactNode[];
}) => {
  const env = getEnv();
  const isProd = isProdEnv();
  const isITLessEnv = ITLess();
  const intl = useIntl();
  const prefix = isProd ? '' : `${env === 'ci' ? 'qa' : env}.`;
  const accountNumberTooltip = `${intl.formatMessage(messages.useAccountNumber)}`;
  const questionMarkRef = useRef(null);
  const { logout } = useContext(ChromeAuthContext);
  const enableMyUserAccessLanding = useFlag('platform.chrome.my-user-access-landing-page');
  const myUserAccessPath = enableMyUserAccessLanding ? '/iam/user-access/overview' : '/iam/my-user-access';

  return [
    <DropdownItem key="Username" isDisabled>
      <dl className="chr-c-dropdown-item__stack">
        <dt className="chr-c-dropdown-item__stack--header">{intl.formatMessage(messages.username)}</dt>
        <dd className="chr-c-dropdown-item__stack--value data-hj-suppress sentry-mask">{username}</dd>
        {isOrgAdmin && <dd className="chr-c-dropdown-item__stack--subValue">{intl.formatMessage(messages.orgAdministrator)}</dd>}
      </dl>
    </DropdownItem>,
    <React.Fragment key="account wrapper">
      {accountNumber && (
        <Tooltip triggerRef={questionMarkRef} id="accountNumber-tooltip" content={accountNumberTooltip}>
          <DropdownItem component="span" key="Account" className="chr-c-disabled-pointer">
            <dl className="chr-c-dropdown-item__stack">
              {!isITLessEnv && (
                <>
                  <dt className="chr-c-dropdown-item__stack--header">
                    {intl.formatMessage(messages.accountNumber)}
                    <span ref={questionMarkRef} className="visible-pointer pf-v6-u-ml-sm">
                      <QuestionCircleIcon />
                    </span>
                  </dt>
                  <dd className="chr-c-dropdown-item__stack--value sentry-mask data-hj-suppress">{accountNumber}</dd>
                </>
              )}
              {isInternal && <dd className="chr-c-dropdown-item__stack--subValue">{intl.formatMessage(messages.internalUser)}</dd>}
            </dl>
          </DropdownItem>
        </Tooltip>
      )}
      {orgId && (
        <DropdownItem key="Org ID" isDisabled ouiaId="chrome-user-org-id">
          <dl className="chr-c-dropdown-item__stack">
            <dt className="chr-c-dropdown-item__stack--header">{intl.formatMessage(messages.orgId)}</dt>
            <dd className="chr-c-dropdown-item__stack--value">{orgId}</dd>
          </dl>
        </DropdownItem>
      )}
    </React.Fragment>,
    <Divider component="li" key="separator" />,
    <React.Fragment key="My Profile wrapper">
      {!isITLessEnv && (
        <DropdownItem
          key="My Profile"
          to={`https://www.${prefix}redhat.com/wapps/ugc/protected/personalInfo.html`}
          target="_blank"
          rel="noopener noreferrer"
          component="a"
        >
          {intl.formatMessage(messages.myProfile)}
        </DropdownItem>
      )}
    </React.Fragment>,
    <React.Fragment key="My user access wrapper">
      <DropdownItem
        component={({ className }) => (
          <ChromeLink className={className} href={myUserAccessPath} appId="rbac">
            {intl.formatMessage(messages.myUserAccess)}
          </ChromeLink>
        )}
        key="My user access"
      />
    </React.Fragment>,
    <React.Fragment key="user prefs wrapper">
      <DropdownItem
        component={({ className }) => (
          <ChromeLink className={className} href="/settings/notifications/user-preferences" appId="userPreferences">
            {intl.formatMessage(messages.userPreferences)}
          </ChromeLink>
        )}
        key="User preferences"
      />
    </React.Fragment>,
    <React.Fragment key="internal wrapper">
      {isInternal && isProd && (
        <DropdownItem
          key="Internal"
          component={({ className }) => (
            <ChromeLink className={className} href="/internal" appId="internal">
              {intl.formatMessage(messages.internal)}
            </ChromeLink>
          )}
        />
      )}
    </React.Fragment>,
    <DropdownItem key="logout" component="button" onClick={logout}>
      {intl.formatMessage(messages.logout)}
    </DropdownItem>,
    extraItems,
  ];
};

export type UserToggleProps = {
  isSmall?: boolean;
  extraItems?: React.ReactNode[];
};

const UserToggle = ({ isSmall = false, extraItems = [] }: UserToggleProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const {
    user: {
      identity: { user, account_number, internal },
    },
  } = useContext(ChromeAuthContext);
  const name = user?.first_name + ' ' + user?.last_name;

  const onSelect = (event: any) => {
    if (['A', 'BUTTON'].includes(event.target.tagName)) {
      setIsOpen(!isOpen);
    }
  };

  const onToggle = () => {
    setIsOpen((prev) => !prev);
  };
  return (
    <Dropdown
      popperProps={{
        position: 'right',
      }}
      aria-label="Overflow actions"
      ouiaId="chrome-user-menu"
      onSelect={onSelect}
      onOpenChange={setIsOpen}
      toggle={(toggleRef) => (
        <MenuToggle
          ref={toggleRef}
          isExpanded={isOpen}
          isFullHeight
          onClick={onToggle}
          variant={isSmall ? 'plain' : undefined}
          className={classNames('data-hj-suppress', 'sentry-mask', { 'pf-v6-u-pr-lg pf-v6-u-pl-lg': isSmall })}
          {...(isSmall
            ? {
                id: 'UserMenu',
                'widget-type': 'UserMenu',
              }
            : {
                icon: <UserIcon />,
              })}
        >
          {isSmall ? <EllipsisVIcon /> : name}
        </MenuToggle>
      )}
      className="chr-c-dropdown-user-toggle"
      isOpen={isOpen}
    >
      <DropdownList>
        {/* Bad PF typings, child nodes can be used */}
        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
        {/* @ts-ignore */}
        <DropdownItems
          username={user?.username}
          isOrgAdmin={user?.is_org_admin}
          accountNumber={account_number}
          orgId={internal?.org_id}
          isInternal={user?.is_internal}
          extraItems={extraItems}
        />
      </DropdownList>
    </Dropdown>
  );
};

// TODO update this to use account_id
export default UserToggle;
