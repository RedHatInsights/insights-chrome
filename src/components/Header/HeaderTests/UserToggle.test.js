import React from 'react';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import UserToggle from '../UserToggle';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import ChromeAuthContext from '../../../auth/ChromeAuthContext';

jest.mock('../UserIcon', () => () => '<UserIcon />');

jest.mock('@unleash/proxy-client-react', () => {
  const actual = jest.requireActual('@unleash/proxy-client-react');
  return {
    __esModule: true,
    ...actual,

    useFlag: () => false,
    useFlagsStatus: () => ({ flagsReady: true }),
  };
});

describe('UserToggle', () => {
  const contextValueMock = {
    user: {
      identity: {
        account_number: 'some accountNumber',
        user: {
          username: 'someUsername',
          first_name: 'someFirstName',
          last_name: 'someLastName',
        },
      },
    },
  };
  it('should render correctly with isSmall false', async () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={configureStore()({ chrome: {} })}>
          <ChromeAuthContext.Provider value={contextValueMock}>
            <UserToggle />
          </ChromeAuthContext.Provider>
        </Provider>
      </MemoryRouter>
    );
    await act(async () => {
      await screen.getByText('someFirstName someLastName').click();
    });
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with isSmall true', () => {
    const { container } = render(
      <ChromeAuthContext.Provider value={contextValueMock}>
        <UserToggle isSmall />
      </ChromeAuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly as org admin', () => {
    const { container } = render(
      <ChromeAuthContext.Provider
        value={{
          user: {
            ...contextValueMock.user,
            is_org_admin: true,
          },
        }}
      >
        <UserToggle />
      </ChromeAuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
