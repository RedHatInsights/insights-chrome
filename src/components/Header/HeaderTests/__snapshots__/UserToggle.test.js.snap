// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserToggle should render correctly as org admin 1`] = `
<div>
  <button
    aria-expanded="false"
    class="pf-v5-c-menu-toggle pf-m-full-height data-hj-suppress sentry-mask"
    data-ouia-component-id="OUIA-Generated-MenuToggle-2"
    data-ouia-component-type="PF5/MenuToggle"
    data-ouia-safe="true"
    type="button"
  >
    <span
      class="pf-v5-c-menu-toggle__icon"
    >
      &lt;UserIcon /&gt;
    </span>
    <span
      class="pf-v5-c-menu-toggle__text"
    >
      someFirstName someLastName
    </span>
    <span
      class="pf-v5-c-menu-toggle__controls"
    >
      <span
        class="pf-v5-c-menu-toggle__toggle-icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`UserToggle should render correctly with isSmall false 1`] = `
<div>
  <button
    aria-expanded="true"
    class="pf-v5-c-menu-toggle pf-m-expanded pf-m-full-height data-hj-suppress sentry-mask"
    data-ouia-component-id="OUIA-Generated-MenuToggle-1"
    data-ouia-component-type="PF5/MenuToggle"
    data-ouia-safe="true"
    type="button"
  >
    <span
      class="pf-v5-c-menu-toggle__icon"
    >
      &lt;UserIcon /&gt;
    </span>
    <span
      class="pf-v5-c-menu-toggle__text"
    >
      someFirstName someLastName
    </span>
    <span
      class="pf-v5-c-menu-toggle__controls"
    >
      <span
        class="pf-v5-c-menu-toggle__toggle-icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </span>
    </span>
  </button>
  <div
    aria-label="Overflow actions"
    class="pf-v5-c-menu chr-c-dropdown-user-toggle"
    data-ouia-component-id="chrome-user-menu"
    data-ouia-component-type="PF5/Dropdown"
    data-ouia-safe="true"
    data-popper-escaped="true"
    data-popper-placement="bottom-end"
    data-popper-reference-hidden="true"
    style="position: absolute; left: 0px; top: 0px; z-index: 9999; opacity: 1; transition: opacity 0ms cubic-bezier(.54, 1.5, .38, 1.11); min-width: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="pf-v5-c-menu__content"
    >
      <ul
        class="pf-v5-c-menu__list"
        role="menu"
      >
        <li
          class="pf-v5-c-menu__list-item pf-m-disabled"
          data-ouia-component-id="OUIA-Generated-DropdownItem-1"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            disabled=""
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                <dl
                  class="chr-c-dropdown-item__stack"
                >
                  <dt
                    class="chr-c-dropdown-item__stack--header"
                  >
                    Username:
                  </dt>
                  <dd
                    class="chr-c-dropdown-item__stack--value data-hj-suppress sentry-mask"
                  >
                    someUsername
                  </dd>
                </dl>
              </span>
            </span>
          </button>
        </li>
        <li
          class="pf-v5-c-menu__list-item chr-c-disabled-pointer"
          data-ouia-component-id="OUIA-Generated-DropdownItem-2"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <span
            class="pf-v5-c-menu__item chr-c-disabled-pointer"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                <dl
                  class="chr-c-dropdown-item__stack"
                >
                  <dt
                    class="chr-c-dropdown-item__stack--header"
                  >
                    Account number:
                    <span
                      class="visible-pointer pf-v5-u-ml-sm"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                        />
                      </svg>
                    </span>
                  </dt>
                  <dd
                    class="chr-c-dropdown-item__stack--value sentry-mask data-hj-suppress"
                  >
                    some accountNumber
                  </dd>
                </dl>
              </span>
            </span>
          </span>
        </li>
        <li
          class="pf-v5-c-divider"
          role="separator"
        />
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-3"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <a
            class="pf-v5-c-menu__item"
            href="https://www.qa.redhat.com/wapps/ugc/protected/personalInfo.html"
            rel="noopener noreferrer"
            role="menuitem"
            tabindex="0"
            target="_blank"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                My profile
              </span>
            </span>
          </a>
        </li>
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-4"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <a
            class="pf-v5-c-menu__item"
            data-quickstart-id="iam_my-user-access"
            data-testid="router-link"
            href="/iam/my-user-access"
          >
            My User Access
          </a>
        </li>
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-5"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <a
            class="pf-v5-c-menu__item"
            data-quickstart-id="settings_notifications_user-preferences"
            data-testid="router-link"
            href="/settings/notifications/user-preferences"
          >
            User Preferences
          </a>
        </li>
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-6"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                Log out
              </span>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`UserToggle should render correctly with isSmall true 1`] = `
<div>
  <button
    aria-expanded="false"
    class="pf-v5-c-menu-toggle pf-m-plain pf-m-full-height data-hj-suppress sentry-mask pf-v5-u-pr-lg pf-v5-u-pl-lg"
    data-ouia-component-id="OUIA-Generated-MenuToggle-plain-1"
    data-ouia-component-type="PF5/MenuToggle"
    data-ouia-safe="true"
    id="UserMenu"
    type="button"
    widget-type="UserMenu"
  >
    <svg
      aria-hidden="true"
      class="pf-v5-svg"
      fill="currentColor"
      height="1em"
      role="img"
      viewBox="0 0 192 512"
      width="1em"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      />
    </svg>
  </button>
</div>
`;
